crates/bevy_asset/src/io/source.rs

```diff
@@ -587,7 +587,7 @@ impl AssetSources {
             AssetSourceId::Name(name) => self
                 .sources
                 .get(&name)
-                .ok_or_else(|| MissingAssetSourceError(AssetSourceId::Name(name))),
+                .ok_or(MissingAssetSourceError(AssetSourceId::Name(name))),
```

Comment:
It's not actually constructed: since it's just wrapping in enum variants, it's basically as simple as a copy anyway, and adding an extra level of function indirection just makes it harder to optimize.

Reply:
What a useful lint :)